libV2/models/Base.js

```diff
@@ -0,0 +1,75 @@
+const Joi = require('@hapi/joi');
+
+class BaseModel {
+    constructor(data) {
+        if (data !== undefined) {
+            this._data = data;
+        } else {
+            this._data = {};
+        }
```

Comment:
Can just use `|| {}` here and it should work the same.
```suggestion
        this._data = data || {};
```